### ISSUE TITLE
TinyMCE Images inserted from MediaLibrary in body content don't display in Rss feeds

### ISSUE THREAD

**Issue by CSurieux** [OrchardCMS/Orchard#6786]
I just discovered that images inserted in Blog-Posts using Open Live Writer are correctly displayed in Rss feeds because that have a full path url.
But when I edit the same Blog-Post in tinyMCE and insert images from the library, they get a relative path (???). 
They are correctly displayed in Orchard, but disappear from rss feed reader.

**Comment by bleroy** [comment]
Yes, this is why I wrote this: https://vandelay.codeplex.com/SourceControl/latest#Filters/RelativeUrlHtmlFilter.cs quite a while ago. RSS is clearly not doing the right thing, but one has to wonder if the OLW behavior is correct: paths relative to the server root would be the only way to ensure that the posts can be moved (using export and import) to a different server with its images, without alterations.

**Comment by CSurieux** [comment]
Shouldn't it be export which should  make relative paths and then import rebuild full paths?

**Comment by donaldboulton** [comment]
Had to create my own sitemap or any .xml  .rss files since 1.9.1, anything .xml just does not work, and ya have to tell the sever not to cache .xml or .rss in web.config
Ignore url's in caching for example `/rss?containerid=533` does not work.
Even with Vandelay Industries remote .rss your get un-display able garbled .rss, from other Orchard sites; a Orchard Issue.
Display as
`<?xml version="1.0" encoding="utf-8"?>`
`<rss version="2.0">`
Should be
`<?xml version='1.0' encoding='UTF-8'?>`
`<rss version="2.0" xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:sy="http://purl.org/rss/1.0/modules/syndication/" xmlns:admin="http://webns.net/mvcb/" xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#" xmlns:content="http://purl.org/rss/1.0/modules/content/">`
Publishing date
`<pubDate>`

Should be
`<dc:date>`

No site description just Blog posts

Should be
`<channel> ` 
   `<title>Test</title>`
    `<link>http://test.com/test.rss</link>`     
    `<image>`
      `<url>http://test.com/orchardburning_200.min.png</url>`
      `<title>Test Site/title>`     
      `<link>http://test.com</link>`
    `</image>`
      `<description>YOUR SITE DESCRIPTION</description>`
      `<dc:language>en-us</dc:language>`
      `<dc:creator>YOUR ADDRESS@mail.com</dc:creator>`
      `<dc:rights>Copyright 2016</dc:rights>`
      `<admin:generatorAgent rdf:resource="http://www.movabletype.org/?v=2.21" />`
      `<admin:errorReportsTo rdf:resource="mailto:YOUR MAIL@mail.com"/>`
// Then Blog Post Links
      `<item>`
        `<title>BLOG PAGE TITLE</title>`
        `<link>BLOG LINK</link>`
        `<description>Anything</description>`
        `<dc:date>Fri, 26 Feb 2016 06:01:00 GMT</dc:date>`
        `<guid ispermalink="true">http:/test.com/photoswipe-isotope</guid>`
        `</item>`
      `<item>`..........
.............................

**Comment by sebastienros** [comment]
The only solution is to provide a service that can transform urls when they are rendered, and we store then in a normalized format like `~/url`. Different cases based on the url format that is used on the client:

FQ url
http://localhost:31321/OrchardLocal/TENANT

=> ~/image.png

Relative url
/Media/Tenant1/image.png

=> ~/Media/Tenant1/image.png

Relative to a tenant prefix
/TENANT/image.png

Relative to current page
foo/image.png

=> unaltered

Prevent any server-side alteration of the url
[TENANT/foo/image.png]

=> TENANT/foo/image.png


In the case of TinyMCE it get FQ urls and sends them back on the server, the server can convert these urls to normalized ones in the database. Any view that renders HTML with urls should use the service to convert the urls to fully qualified ones.
